docs/Authenticating-Users-via-keycloak-SAML.md

```diff
@@ -105,9 +102,37 @@ keytool -genkey -alias secure-key -keyalg RSA -keystore samlKeystore.jks
     saml.idp.metadata.attribute.role=roles
     saml.custom.userservice.class=org.cbioportal.security.spring.authentication.keycloak.SAMLUserDetailsServiceImpl
     # global logout (as opposed to local logout):
-    saml.logout.url=/login.jsp?logout_success=true
+    saml.logout.local=false
+    saml.logout.url=/
 ```
 
+## Integrating keycloak with your company-wide authentication service
+
+When integrating keycloak with your company-wide authentication service, the user credentials will reside at your institute's users directory and keycloak
+will work as a proxy between your LDAP (or SAML) service and cBioPortal.
+
+Please refer to keycloack documentation [http://www.keycloak.org/documentation.html](http://www.keycloak.org/documentation.html) for more information on how to integrate keycloak with your local LDAP/SAML service.
+
+## Authorization with keycloak
+
+### Local keycloak users: create users and assigning roles
+
+1. To create a user, click on Users in the left menu bar. This menu option brings you to the user list page. On the right side of the empty user list,
+ you should see an Add User button. Click that to start creating your new user.
+![](images/previews/add-user.png)
+2. Next, assign roles to each user through the Role Mappings tab for that single user.
+**A composite role is a role that has one or more additional roles associated with it**. When a composite role is mapped to the user,
+ the user also gains the roles associated with that composite. This inheritance is recursive so any composite of composites also gets inherited.
+![](images/previews/assign-role-to-user.png)
+
+### Institute's IDP users: assigning roles
+
+When integrating keycloak with your company-wide authentication service, the user credentials will reside at your institute's users directory and keycloak
+will work as a proxy between your LDAP (or SAML) IDP service and cBioPortal. Nevertheless, the authorization roles
+should still be configured for the users. The steps below detail how to do this.
+
+TODO
```

Comment:
@fedde-s maybe you can also help fill in this part once you start testing with keycloak?